packages/web/src/scss/components/SegmentedControl/_SegmentedControlItem.scss

```
@@ -0,0 +1,112 @@
+// 1. JS version is used when there is need for more items then $max-items. CSS variable is generated by JS.
+// 2. In JS version, there is toggle of opacity because of transition effect on initialization (active item is not always the last one)
+// 3. Label has z-index 2, so it's always on top of the highlight, last one contains pseudo element with highlight, so checked label has z-index 3.
+// 4. Because of truncate on the text inside of label we need to set min-width, otherwise flex will overflow and all items will not have the same width.
+
+@use '@tokens' as tokens;
+@use '../../tools/accessibility';
+@use '../../tools/dictionaries';
+@use '../../tools/typography';
+@use 'theme';
+@use 'tools';
+
+.SegmentedControlItem__label {
+    @include typography.generate(theme.$item-label-typography);
+
+    position: relative;
+    z-index: 2;
+    display: flex;
+    flex: 1 0 0;
+    column-gap: theme.$item-gap;
+    align-items: center;
+    justify-content: center;
+    min-width: theme.$item-min-width; // 4.
+    height: theme.$item-height;
+    padding-inline: theme.$item-padding-y;
+    padding-block: theme.$item-padding-x;
+    border-radius: theme.$item-border-radius;
+    user-select: none;
+    cursor: pointer;
+    transition: inherit;
+
+    @media (hover: hover) {
+        :not(:checked, :disabled) + &:hover {
+            background-color: theme.$item-background-hover;
+        }
+    }
+
+    &::after {
+        content: '';
+        position: absolute;
+        inset: 0;
+        z-index: -1;
+        border: theme.$item-border-default;
+        border-radius: theme.$item-border-radius;
+        transform: translateX(
+            calc(
+                var(--#{tokens.$css-variable-prefix}segmented-control-highlight-pos, 0) *
+                    -100% - calc(
+                        var(--#{tokens.$css-variable-prefix}segmented-control-highlight-pos, 0) * #{theme.$gap}
+                    )
+            )
+        );
+        pointer-events: none;
+        transition: inherit;
+    }
+}
+
+.SegmentedControlItem__label > svg {
+    flex-shrink: 0;
+}
+
+.SegmentedControlItem__input {
+    @include accessibility.hide-text();
+
+    &:focus-visible + .SegmentedControlItem__label {
+        box-shadow: theme.$focus-ring;
+    }
+
+    &:disabled + .SegmentedControlItem__label {
+        color: theme.$item-disabled-color;
+        cursor: theme.$item-disabled-cursor;
+    }
+
+    &:checked + .SegmentedControlItem__label {
+        z-index: 3; // 3.
+        color: var(--#{tokens.$css-variable-prefix}segmented-control-item-color);
+    }
+}
+
+@include dictionaries.generate-fill-variants(
+    $class-name: 'SegmentedControl',
+    $dictionary-values: theme.$fill-variant-dictionary,
+    $config: theme.$fill-variant-active-config,
+    $custom-component-infix: 'item'
+);
+
+// stylelint-disable selector-max-specificity
```

Please always provide a reason to explain why you decided to disable Stylelint (or ESlint).

For example:

```suggestion
// stylelint-disable selector-max-specificity -- 5.
```